### PR TITLE
#331 CpsDialogService use applicationref instead of viewcontainerref and provide in root

### PR DESCRIPTION
**Pull request by lukasmatta**

### CODE REVIEW

File: projects/cps-ui-kit/src/lib/services/cps-dialog/cps-dialog.service.ts [hunk 1]
```diff
@@ -83,10 +84,17 @@ export class CpsDialogService {
       sub.unsubscribe();
     });
 
-    const componentRef = this.viewContainerRef.createComponent(
-      CpsDialogComponent,
-      { injector: new DynamicDialogInjector(this.injector, map) }
```
Comment (fateeand) [inline projects/cps-ui-kit/src/lib/services/cps-dialog/cps-dialog.service.ts:88]: `map` is not needed anymore

File: projects/cps-ui-kit/src/lib/services/cps-dialog/cps-dialog.service.ts
Comment (fateeand) [inline projects/cps-ui-kit/src/lib/services/cps-dialog/cps-dialog.service.ts:1]: Let's also add `clear()` or `closeAll()` method to the service in order to close all opened dialogs

File: projects/cps-ui-kit/src/lib/services/cps-dialog/cps-dialog.service.ts (hunk 1 above)
Comment (lukasmatta) [reply projects/cps-ui-kit/src/lib/services/cps-dialog/cps-dialog.service.ts:88]: good catch, removed the map

File: projects/cps-ui-kit/src/lib/services/cps-dialog/cps-dialog.service.ts
Comment (lukasmatta) [reply projects/cps-ui-kit/src/lib/services/cps-dialog/cps-dialog.service.ts:1]: Added `closeAll()` method

File: projects/composition/src/app/pages/dialog-page/dialog-page.component.html
Comment (fateeand) [inline projects/composition/src/app/pages/dialog-page/dialog-page.component.html:1]: I would suggest using `outlined` buttons with `remove` icon, separated from other dialog-opening buttons by `border-bottom: 1px solid lightgrey;`

e.g.:
Screenshot 2024-03-18 at 18 53 33

Comment (lukasmatta) [reply projects/composition/src/app/pages/dialog-page/dialog-page.component.html:1]: Changed button type and added border

**Review by fateeand — APPROVED** [review]

**Pull request merged**